packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -1122,4 +1119,438 @@ describe('KeyringController', () => {
+        { data: JSON.stringify(msgParams), from: account },
+        SignTypedDataVersion.V4,
+      );
+      const recovered = recoverTypedSignature({
+        data: msgParams,
+        signature,
+        version: SignTypedDataVersion.V4,
+      });
+      expect(account).toBe(recovered);
+    });
+
+    it('should forget Ledger Keyring', async () => {
+      const accounts = await ledgerKeyring.getAccounts();
+      expect(accounts).toHaveLength(1);
+
+      await keyringController.forgetLedger();
+      expect(keyringController.state.keyrings[1].accounts).toHaveLength(0);
+    });
+
+    it('should return Ledger Keyring for corresponding account', async () => {
+      const ledgerAccount = '0xe908e4378431418759b4f87b4bf7966e8aaa5cf2';
+
+      const keyring = await keyringController.getAccountKeyringType(
+        ledgerAccount,
+      );
+
+      expect(keyring).toBe(KeyringTypes.ledger);
+    });
+
+    it('should connect to ledger hardware', async () => {
+      const setTransportStub = sinon.stub(ledgerKeyring, 'setTransport');
+      setTransportStub.resolves();
+
+      const getAppAndVersionStub = sinon.stub(
+        ledgerKeyring,
+        'getAppAndVersion',
+      );
+
+      const mockedAppVersionValue = {
+        appName: 'mockApp',
+        version: 'mockVersion',
+      };
+      getAppAndVersionStub.resolves(mockedAppVersionValue);
+
+      const mockDeviceId = 'mockDeviceId';
+      const mockTransport = new MockTransport();
+
+      const appName = await keyringController.connectLedgerHardware(
+        mockTransport,
+        mockDeviceId,
+      );
+
+      expect(appName).toBe(mockedAppVersionValue.appName);
+    });
+
+    it('should open ethereum app on ledger', async () => {
+      const openEthereumAppOnLedgerStub = sinon.stub(
+        ledgerKeyring,
+        'openEthApp',
+      );
+      openEthereumAppOnLedgerStub.resolves();
+      await keyringController.openEthereumAppOnLedger();
+      expect(openEthereumAppOnLedgerStub.callCount).toBe(1);
+    });
+
+    it('should close running ledger app on ledger', async () => {
+      const getAppAndVersionStub = sinon.stub(
+        ledgerKeyring,
+        'getAppAndVersion',
+      );
+
+      const mockedAppVersionValue = {
+        appName: 'mockApp',
+        version: 'mockVersion',
+      };
+
+      const mockedAppVersionValueBolos = {
+        appName: 'BOLOS',
+        version: 'mockVersion',
+      };
+
+      getAppAndVersionStub.onCall(0).resolves(mockedAppVersionValue);
+      getAppAndVersionStub.onCall(1).resolves(mockedAppVersionValueBolos);
+
+      const quitAppSpy = sinon.stub(ledgerKeyring, 'quitApp');
+      quitAppSpy.resolves();
+
+      await keyringController.closeRunningAppOnLedger();
+      expect(quitAppSpy.callCount).toBe(1);
+
+      // Testing the path when app name is not BOLOS
+      await keyringController.closeRunningAppOnLedger();
+      expect(quitAppSpy.callCount).toBe(1);
```

Comment:
The comment here seems to be misplaced, and the second assertion doesn't seem to be correct. The first time `closeRunningAppOnLedger` is called, `getAppAndVersion` should return an object with an `appName` that is not 'BOLOS', and the second time, it should return an object with an `appName` that _is_ 'BOLOS'. Therefore, I would think this would be 

``` suggestion
      // appName is not BOLOS
      await keyringController.closeRunningAppOnLedger();
      expect(quitAppSpy.callCount).toBe(1);

      // appName is BOLOS
      await keyringController.closeRunningAppOnLedger();
      expect(quitAppSpy.callCount).toBe(0);
```

Can you investigate why this might be?

Reply:
Yep your right the comment was misplaced. For the second call, the callCount is still 1 because the `quitAppSpy` saved the history of the previous `closeRunningAppOnLedger` call, the number of calls didn't increment. 

I can add a quitAppSpy.reset() to reset the counter, and the expected callCount will be 0 when testing for BOLOS.